src/storage-types/src/collections.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package mz_storage_types.collections;
+
+message GlobalId {
+    oneof value {
+        uint64 system = 1;
+        uint64 user = 2;
+        uint64 transient = 3;
+        google.protobuf.Empty explain = 4;
+    }
+}
```

Comment:
I don't love having to duplicate `GlobalId`, but I don't know of a way to prevent duplicating it, without complicating our protobuf snapshotting logic

Reply:
It was already duplicated in `repr`, https://github.com/MaterializeInc/materialize/blob/main/src/repr/src/global_id.proto, now it's triplicated. I think the better solution would be to update the other proto messages in this file to use the `repr` message instead of this one. However, unless there's a big push to do that in this PR, I'll leave that to the storage folks to clean up since there's already a bunch of other duplicate messages in this crate.

Reply:
Ahh I didn't realize that, works for me!